src/upload.cpp

```diff
@@ -110,6 +115,180 @@ void Upload::onTeleTANFinished()
     reply->deleteLater();
 }
 
+void Upload::uploadGUID(QString const &guid)
+{
+    if (((m_status == StatusIdle) || (m_status == StatusError)) && (m_reply == nullptr)) {
+        bool valid = validateGUID(guid);
+        if (valid) {
+            setStatus(StatusSubmitGUID);
+            submitGUID(guid);
+        }
+        else {
+            qDebug() << "Invalid GUID";
+            setStatusError(ErrorInvalidGUID);
+        }
+    }
+    else {
+        // Do nothing until the existing upload completes
+        qDebug() << "Upload already in progress";
+    }
+}
+
+void Upload::submitGUID(QString const &guid)
+{
+    QCryptographicHash hasher(QCryptographicHash::Sha256);
+    hasher.addData(guid.toLatin1());
+    QString hashedGuid = static_cast<QString>(hasher.result());
+
+    if ((m_status == StatusSubmitGUID) && (m_reply == nullptr) && (hashedGuid.length() > 0)) {
+        QNetworkRequest request;
+
+        request.setUrl(QUrl(AppSettings::getInstance().verificationServer() + "/version/v1/registrationToken"));
+        request.setRawHeader("accept", "*/*");
+        request.setRawHeader("Content-Type", "application/json");
+        request.setRawHeader("cwa-fake", "0");
+
+        QJsonObject body;
+        body.insert("keyType", "hashedGUID");
+        body.insert("key", hashedGuid);
+        QJsonDocument doc(body);
+        QByteArray data = doc.toJson();
+        m_reply = m_manager->post(request, data);
+
+        connect(m_reply, &QNetworkReply::finished, this, &Upload::onGUIDFinished);
+    }
+    else {
+        qDebug() << "Incorrect state attempting to submit GUID";
+        setStatusError(ErrorInternal);
+    }
+}
+
+void Upload::onGUIDFinished()
+{
+    QNetworkReply *reply;
+    QByteArray data;
+    QJsonParseError error;
+    QJsonDocument doc;
+
+    reply = m_reply;
+    m_reply = nullptr;
+
+    switch (reply->error()) {
+    case QNetworkReply::NoError:
+        data = reply->readAll();
+
+        doc = QJsonDocument::fromJson(data, &error);
+        if (doc.isNull()) {
+            qDebug() << "JSON error: " << error.errorString();
+            setStatusError(ErrorParsing);
+        }
+        else {
+            QString regToken = doc.object().value("registrationToken").toString();
+            if (regToken.isEmpty()) {
+                qDebug() << "Invalid registration token";
+                setStatusError(ErrorInvalidRegToken);
+            }
+            else {
+                if (SFSecretHelper::getInstance().store(REGTOKEN_KEY, regToken.toLatin1()).returncode == 0) {
```

Comment:
Running against the test sever, when it gets to here the following error is triggered:
```
[D] onStatusChanged:16 - guidPage status: 3
[W] SFSecretHelper::store:61 - Failed to create collection  "harbour-contrac"  for storing the key!
[W] SFSecretHelper::store:62 - Error: Sailfish::Secrets::Result::ErrorCode(InvalidCollectionError) "SQLCipher plugin only supports collection names with alphanumeric Latin-1 characters"
[D] Upload::onGUIDFinished:211 - SFSecret couldn't store regtoken
[D] Upload::setStatusError:503 - Setting error:  9
```